Contest_API.md

```diff
@@ -1571,65 +1571,74 @@ If the client fails to respond to multiple requests over a period of
+| ---------------- | ---------------- | --------- | ---------- | -------------------------------------------------------------------------------------------------------------------- |
+| `/webhooks`      | application/json | yes       | CCS        | JSON array of all webhook callbacks with elements as defined in the table below. Also used to register new webhooks. |
+| `/webhooks/<id>` | application/json | yes       | CCS        | JSON object of a single webhook callback with elements as defined in the table below                                 |
 
-JSON elements of webhook objects:
+JSON elements of webhook callback objects:
 
-| Name | Type   | Required? | Nullable? | Description           |
-| ---- | ------ | --------- | --------- | --------------------- |
-| url  | string | yes       | no        | The url for callbacks |
+| Name       | Type            | Required? | Nullable? | Description                                                                           |
+| ---------- | --------------- | --------- | --------- | ------------------------------------------------------------------------------------- |
+id           | ID              | yes       | no        | identifier of the webhook.                                                            |
+url          | string          | yes       | no        | The URL to post HTTP callbacks to.                                                    |
+endpoints    | array of string | yes       | no        | Names of endpoints to receive callbacks for. Empty array means all endpoints.         |
+contest\_ids | array of ID     | yes       | no        | ID’s of contests to receive callbacks for. Empty array means all configured contests. |
+active       | boolean         | yes       | no        | Whether the webhook is enabled.                                                       |
 
-TODO: include filter details?
+The idea behind the `active` field is that if the CCS decides a webhook isn’t behaving and wants to disable sending callbacks to it,
+it can update this active field. It also allows one to temporarily disable a webhook using some UI if a CCS implements it.
 
 ##### Adding a webhook
 
-To register a webhook, you need to post your server's callback url. The
-general format to register a webhook is:
+To register a webhook, you need to post your server's callback URL.
+To do so, perform a `POST` request with a JSON body with the fields (except `id`) from the above table to the `/webhooks` endpoint together with one additional field,
+called `token`. In this field put a client-generated token that can be used to verify that callbacks come from the CCS.
 
-```json
-{"url": "<callback url>", "auth": ... }
+```note
+TODO: What about using sane defaults for `endpoints`, `contest_ids` and `active` when adding a new webhook?
```

Comment:
The only possible (?) sane default would be empty. That's not a great default for an online setting though (but it works pretty great at an onsite event.)

Maybe `contest_ids` should be required? (At least if `/webhooks` is at the top level (see earlier comment).

Reply:
We will use empty arrays as defaults for `contest_ids` and `endpoints`.

We will remove `active` for now from the webhooks.